xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/test/java/org/xwiki/tool/packager/PackageMojoTest.java

```diff
@@ -57,4 +81,35 @@ public void replaceProperty()
             + "whatever";
         assertEquals(expected, mojo.replaceProperty(content, context));
     }
+
+    @Test
+    void before() throws Exception
+    {
+        MavenProject project = mock(MavenProject.class);
+        RepositorySystem repositorySystem = mock(RepositorySystem.class);
+        Artifact artifact = mock(Artifact.class);
+
+        when(project.getProperties()).thenReturn(new Properties());
+        when(repositorySystem.resolve(any())).thenReturn(new ArtifactResolutionResult());
+        when(repositorySystem.createArtifact(any(), any(), any(), any(), any())).thenReturn(artifact);
+        // Return a malformed jar file with paths containing '../' to trigger the zip slip check.
+        File jar = Paths.get(PackageMojoTest.class.getResource("/evil.jar").toURI()).toFile();
+        when(artifact.getFile()).thenReturn(jar);
+
+        PackageMojo mojo = new PackageMojo();
+        // Mock the logs to prevent the quality checks to fail because of unexpected logs.
+        mojo.setLog(mock(Log.class));
+        ReflectionUtils.setFieldValue(mojo, "hibernateConfig", mock(File.class));
+        ReflectionUtils.setFieldValue(mojo, "project", project);
+        ReflectionUtils.setFieldValue(mojo, "repositorySystem", repositorySystem);
+        // Initialize a temporary directory otherwise the configuration files are added to the project folders.
+        ReflectionUtils.setFieldValue(mojo, "outputPackageDirectory", this.tmpDir);
+
+        MojoExecutionException exception = assertThrows(MojoExecutionException.class, mojo::before);
+        assertEquals(IOException.class, exception.getCause().getClass());
+        assertEquals(String.format(
+                "Template [%1$s/webapps/xwiki/WEB-INF/../test] is outside of the configuration file directory "
```

Comment:
what is what you call the "configuration file directory"? Are you referring to the webapp `WEB-INF`directory?

Reply:
> what is what you call the "configuration file directory"? Are you referring to the webapp `WEB-INF`directory?

Yes, I reused the vocabulary used in `generateConfigurationFiles` but a better term could exist :)

> Note: I would create the JAR as part of the test since it's not great to include a binary that we cannot easily recreate and it's easy to crate a jar in the test.

I'll see if I can produce a "malformed" zip programmatically.